river/utils/math.py

```diff
@@ -353,3 +354,27 @@ def woodbury_matrix(A: np.ndarray, U: np.ndarray, V: np.ndarray):
     eye = np.eye(len(V))
     Au = A @ U
     A -= Au @ np.linalg.inv(eye + V @ Au) @ V @ A
+
+
+def log_sum_2_exp(a: float, b: float) -> float:
+    """Computation of log( (e^a + e^b) / 2) in an overflow-proof way
+
+    Parameters
+    ----------
+    a : float
+        First number
+
+    b : float
+        Second number
+
+    Returns
+    -------
+    output : float
+        Value of log( (e^a + e^b) / 2) for the given a and b
+    """
+    # TODO: if |a - b| > 50 skip
+    # TODO: try several log and exp implementations
```

Comment:
Is this still valid?

Reply:
This function is actually imported from Onelearn's implementation and would need review by a Computer Scientist maybe? I think it's meant to compute sums of exponentials without facing an Overflow Exception (see when the trees update) @smastelini 
If you think it's not relevant we can remove it =)

Reply:
My bad, I meant the # TODO in the code :D

Reply:
Oh, that's what I meant too. These TODO are out of my reach tbh, so I kept them here in case somebody knows what it would mean to skip the |a - b| > 50 and different kinds of log & exp implementation to avoid the Overflow issue and still be able to compute the expected value.
I guess we can remove the TODO, but it might be good to remember the goal of this function and that it could apparently be improved?

Reply:
Oh, I see! Let's keep them, no problem :)